src/base/misc/disks.c

```diff
@@ -582,6 +590,10 @@ static void hdisk_auto(struct disk *dp)
       error("Hmm... BLKSSZGET failed (not fatal): %s\n", strerror(errno));
       sector_size = SECTOR_SIZE;
     }
+    if (dp->num_secs & (sector_size - 1) ) {
```

Comment:
What is this?
Something is wrong for sure.

Reply:
I don't know what you mean.

Reply:
Why looking for reminder of num_secs %
sector_size? Usually you look for reminder
of image_size % sector_size, which looks
much better.

Reply:
In this case num_secs here holds the block device's size in bytes.